ci-operator/step-registry/gather/alibabacloud-console/gather-alibabacloud-console-commands.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+fi
+
+if test -f "${SHARED_DIR}/alibaba-instance-ids.txt"
+then
+
+	export ALIBABA_CLOUD_CREDENTIALS_FILE="${SHARED_DIR}/alibabacreds.ini"
+	export ALIBABA_CLI_CREDENTIALS_FILE="${SHARED_DIR}/config"
+
+	pushd /tmp
+	wget https://aliyuncli.alicdn.com/aliyun-cli-linux-latest-amd64.tgz -O aliyun-cli.tgz
+	tar zxvf aliyun-cli.tgz
+	popd
+
+	/tmp/aliyun version
+
+	echo "Settting --config-path=${ALIBABA_CLI_CREDENTIALS_FILE} and --region=${LEASED_RESOURCE}"
+	/tmp/aliyun --config-path "${ALIBABA_CLI_CREDENTIALS_FILE}" configure set --region "${LEASED_RESOURCE}"
+
+
+	cat "${SHARED_DIR}/alibaba-instance-ids.txt" >> "${TMPDIR}/node-provider-IDs.txt"
+else
+
+	echo "No alibaba-instance-ids.txt; skipping console log retrieval."
+        exit 0
+fi
+
+
+cat "${TMPDIR}/node-provider-IDs.txt" | sort | grep . | uniq | while read -r INSTANCE_ID
+do
+	echo "Gathering console logs for ${INSTANCE_ID}"
+        /tmp/aliyun --config-path "${ALIBABA_CLI_CREDENTIALS_FILE}" ecs GetInstanceConsoleOutput --RegionId "${LEASED_RESOURCE}" --InstanceId "$INSTANCE_ID" | jq -r '.ConsoleOutput' | base64 -d > "${ARTIFACT_DIR}/${INSTANCE_ID}" &
+	wait "$!"
```

Comment:
Not sure where the ID came from, but [the rehearsal][1] failed:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/20841/rehearse-20841-pull-ci-openshift-installer-master-e2e-alibaba/1485730605567053824/build-log.txt | grep gather-alibabacloud-console
INFO[2022-01-24T22:56:36Z] Running step e2e-alibaba-gather-alibabacloud-console. 
INFO[2022-01-24T22:57:06Z] Logs for container test in pod e2e-alibaba-gather-alibabacloud-console: 
INFO[2022-01-24T22:57:06Z] Step e2e-alibaba-gather-alibabacloud-console failed after 30s. 
Link to job on registry info site: https://steps.ci.openshift.org/job?org=openshift&repo=installer&branch=master&test=e2e-alibaba, "e2e-alibaba" post steps failed: "e2e-alibaba" pod "e2e-alibaba-gather-alibabacloud-console" failed: the pod ci-op-cm47ntwf/e2e-alibaba-gather-alibabacloud-console failed after 20s (failed containers: test): ContainerFailed one or more containers exited
Link to step on registry info site: https://steps.ci.openshift.org/reference/gather-alibabacloud-console
```

[with][2]:

```
Gathering console logs for i-0xih3nnep3txd8c4asyo
[1;31mERROR: SDK.ServerError
ErrorCode: InvalidParameter
Recommend: https://error-center.aliyun.com/status/search?Keyword=InvalidParameter&source=PopGw
RequestId: 78DC8847-01E5-3D1C-BB4A-6F9D2E4C247D
Message: The instanceId provided is not valid
[0m
```

Maybe a bootstrap node?  Maybe we need to accept failures here for the subset that are "I don't recognize that instance ID"?

I also don't understand the empty `tmp-censor254417089` file [here][3].

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20841/rehearse-20841-pull-ci-openshift-installer-master-e2e-alibaba/1485730605567053824
[2]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20841/rehearse-20841-pull-ci-openshift-installer-master-e2e-alibaba/1485730605567053824/artifacts/e2e-alibaba/gather-alibabacloud-console/build-log.txt
[3]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20841/rehearse-20841-pull-ci-openshift-installer-master-e2e-alibaba/1485730605567053824/artifacts/e2e-alibaba/gather-alibabacloud-console/artifacts/

Reply:
I believe this instance ID is the bootstrap. Since it gets torn down the console log query fails. Will attempt to fix this morning.